### PR TITLE
Feature/enable basic and session auth

### PR DESCRIPTION
**Pull request by Sakshi-dell**
# Description
Added a parameter "isBasicAuth" in values.yaml in CSI-PowerScale which can be used to choose between Basic Auth and Session-based Auth.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/139 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Ran unit and Integration tests
- [x] Ran cert-csi sanity with both the authentication methods

### CODE REVIEW

**Review by randeepsharma — DISMISSED** [review]
LGTM

File: service/step_defs_test.go [hunk 1]
```diff
@@ -1862,6 +1863,69 @@ func (f *feature) aIsilonServiceWithParams(user, mode string) error {
 	return nil
 }
 
+func (f *feature) aIsilonservicewithIsiAuthTypeasfalse() error {
```
Comment (bpjain2004) [inline service/step_defs_test.go:1866]: is this still a valid test ?

File: helm/csi-isilon/values.yaml [hunk 2]
```diff
@@ -210,6 +210,13 @@ endpointPort: 8080
 # Default value: false
 skipCertificateValidation: true
 
+# isiAuthType: Indicates whether the authentication will be session-based or basic.
+# Allowed values:
+#   0: enables basic Authentication
+#   1: enables session-based Authentication
+# Default value: 0
+isiAuthType: 0
+
```
[the remark this answers is not in the document]
Comment (Sakshi-dell) [reply helm/csi-isilon/values.yaml:219]: It is already present on  "PLATFORM ATTRIBUTES"

**Review by randeepsharma — APPROVED** [review]

**Review by bpjain2004 — APPROVED** [review]

File: service/service.go [hunk 3]
```diff
@@ -86,6 +86,7 @@ type Opts struct {
 	KubeConfigPath            string
 	allowedNetworks           []string
 	MaxVolumesPerNode         int64
+	IsiAuthType               uint
```
[the remark this answers is not in the document]
Comment (Sakshi-dell) [reply service/service.go]: No, updated in latest commit

File: service/service.go [hunk 4]
```diff
@@ -117,6 +118,7 @@ type IsilonClusterConfig struct {
 	IsiPath                   string `yaml:"isiPath,omitempty"`
 	IsiVolumePathPermissions  string `yaml:"isiVolumePathPermissions,omitempty"`
 	IsDefault                 *bool  `yaml:"isDefault,omitempty"`
+	IsiAuthType               uint   `yaml:"isiAuthType,omitempty"`
```
[the remark this answers is not in the document]
Comment (Sakshi-dell) [reply service/service.go]: Updated in latest commit, removed it from here.

File: service/step_defs_test.go (hunk 1 above)
Comment (Sakshi-dell) [reply service/step_defs_test.go]: The test is valid but the function name needed an update

**Pull request merged**